routemodel/data_retrieval/get_weather.py

```diff
@@ -1,60 +1,69 @@
 ONE_CALL_BASE = 'https://api.openweathermap.org/data/2.5/onecall?'
-PATH = os.path.join(os.path.dirname(__file__), '..', 'routes\ASC2021\ASC2021_draft.csv')
-
-def get_weather(path):
-    
-    weather_api_result = []
-    with open(path, 'r') as wea_2021:
-        for line in wea_2021:
-            row = line.split(',')
-            lat = row[0]
-            lon = row[1].strip()
-
-            query_url = ONE_CALL_BASE + 'lat=' + lat + '&lon=' + lon + '&exclude=hour,daily&units=metric&appid=' + WEATHER_API_KEY
-
-            response = requests.get(query_url)
-            if response.status_code == 200:
-                weather_data = response.json()
-
-                precipitation = 0 if "rain" not in weather_data["current"]["weather"][0] else weather_data["current"]["weather"][0]["rain"]
-
-                weather_api_result.append([
-                    float(lat), float(lon), 
-                    weather_data["current"]["temp"],
-                    weather_data["current"]["wind_speed"],
-                    weather_data["current"]["wind_deg"],
-                    weather_data["current"]["weather"][0]["main"],
-                    weather_data["current"]["weather"][0]["description"],
-                    precipitation
-                ])
-    return weather_api_result
-
-def save_weather_to_csv(weather_api_data):
-    with open('new_get_weather.csv', 'w', newline='') as f:
-        data = csv.writer(f, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
-
-        data.writerow([
-            'Latitude', 
-            'Longitude', 
-            'Temperature (C)', 
-            'Wind Speed (m/s)', 
-            'Wind Direction', 
-            'Weather', 
-            'Weather Description', 
-            'Precipitation (mm)'
-        ])
-
-        for row in weather_api_data:   
-            data.writerow(row)
+PATH = os.path.join(os.path.dirname(__file__), '..', 'routes/ASC2021/ASC2021_draft.csv')
+
+
+def get_weather(lat, long, weather_df):
+    """
+    Calls OpenWeather API to get wind data and returns data in a pandas dataframe
+    @param lat: string containing latitude value
+    @param long: string containing longitude value
+    @param weather_df: dataframe containing the weather data
+    @return: dataframe containing the weather data
+
+    """
+    units = "metric"
+    exclude = "minutely,hourly,daily,alerts"  # To exclude certain weather reports, right now just using current
+    url = ONE_CALL_BASE + "lat={}&lon={}&exclude={}&units={}&appid={}".format(lat, long, exclude, units,
+                                                                              WEATHER_API_KEY)
+    response = requests.get(url)
+
+    if response.status_code == 200:
+        weather_data = response.json()
+
+        precipitation = 0 if "rain" not in weather_data["current"]["weather"][0] else \
+            weather_data["current"]["weather"][0]["rain"]
+        weather_df = weather_df.append({'Latitude': lat,
```

Comment:
some really change spacing here, can we start the dictionary `{` on a new line and then align all the keys?

Reply:
Yep! Don't know why the formatting got so skewed in my commit